core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedService.java

```diff
@@ -300,7 +304,14 @@ private Object invoke(ServiceRequestContext ctx, HttpRequest req,
         try (SafeCloseable ignored = ctx.push()) {
             final ResolverContext resolverContext = new ResolverContext(ctx, req, aggregatedRequest);
             final Object[] arguments = AnnotatedValueResolver.toArguments(resolvers, resolverContext);
-            return method.invoke(object, arguments);
+            if (CoroutineUtil.isSuspendingFunction(method)) {
+                return CoroutineUtil.invokeSuspendingFunction(
```

Comment:
Hello, @okue.
This looks good and Kotlin users could use Armeria well with the Coroutine after this PR is merged.
However, this will bring additional dependencies to the users who don't use Kotlin.
So, we have to make a dedicated module for this purpose.
Then, if the dependency is in the class path, we can enable Coroutine support.

This is one of the approaches we do for checking the dependencies:
https://github.com/line/armeria/blob/master/spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/internal/spring/ArmeriaConfigurationUtil.java#L127

This is an example to call a method using `MethodHandles`
https://github.com/line/armeria/blob/master/spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/WebOperationService.java#L96

Please note that the method handle is initialized only once, so that we don't do expensive operations every time we call `invoke`.

Reply:
Thanks for a nice advice 🙏 
Tests failed due to the problem you mentioned.

Reply:
What I meant was that making a separate module not a package so that we can put
```
implementation "org.jetbrains.kotlinx:kotlinx-coroutines-jdk8"
implementation "org.jetbrains.kotlin:kotlin-reflect"
```
these dependencies into that module so that `core` module does not depend on Kotlin. 😄 
We might name the module as `kotlin` /cc @trustin, @ikhoon